### DISCUSSION TITLE
Reason for "Have full, unrestricted access to Thunderbird, and your computer" permission?

### DISCUSSION THREAD

**Discussion by ctorgalson** [kewisch/gdata-provider#676] [Support] [answered]
This is an awfully broad permission, and I just wanted to learn the reason for it. Is it that the extension needs to be able to descend into directories to use/attach files to calendars?

**Comment by SirIntellegence** [comment]
I am hesitant to install this unless someone can justify the permissions.

**Reply by SirIntellegence** [reply]
I managed to find something that may explain this. I haven't checked code (since I don't have time to do that), but it looks like the addon may be using [old API's](https://support.mozilla.org/en-US/kb/permission-request-messages-thunderbird-extensions#w_have-full-unrestricted-access-to-thunderbird-and-your-computer)). Depending on how old this extension is, rewriting it to the newer API's may not be feasible, but, as I said, I haven't checked. Hopefully, someone else can give a more official answer.

**Comment by kewisch** [comment, marked as answer]
Hey, yes, @SirIntellegence is correct. I'm using the legacy APIs simply because there are no new calendar APIs in Thunderbird yet. In practice I don't access anything other than Thunderbird itself, so there is nothing to worry about.
